### PR TITLE
JDK-8227540: Add README.md and CONTRIBUTING.md to jfx repo

### PR DESCRIPTION
**Pull request by kevinrushforth**
[JDK-8227540](https://bugs.openjdk.java.net/browse/JDK-8227540)

This PR adds a README.md and CONTRIBUTING.md file to the top of the repo in support of moving to GIT as part of project Skara.

These two files already exist in the [.github](https://github.com/javafxports/openjdk-jfx/tree/develop/.github) directory of this GitHub sandbox, which means that *that* README.md file is displayed on the main page of the [javafxports/openjdk-jfx](https://github.com/javafxports/openjdk-jfx) sandbox.

Details of this proposed change is as follows:

The contents of the two new files will be similar to the existing files in the .github dir of the sandbox, but will differ in a few key points. For example, it will remove the text about it being a mirror repository. Eventually, we will add some information about using the Skara tooling, but for this PR I just added a note that the need for sending the RFR email will likely go away.

The .github/README.md file takes precedence over README.md, which allows us to push these changes without disturbing the existing content for the sandbox repo, which will make for a smoother transition.

The new README.md (which will point to the new CONTRIBUTING.md file) will be seen when looking at the [openjdk/jfx](https://github.com/openjdk/jfx) repo.

The existing .github/README.md file (which points to the .github/CONTRIBUTING.md file) will supersede the top level README.md file when looking at the [javafxports/openjdk-jfx](https://github.com/javafxports/openjdk-jfx) sandbox repo.

You can see that by comparing my `JDK-8227540-README` branch for this enhancement in the following two repos: [kevinrushforth/openjdk-jfx](https://github.com/kevinrushforth/openjdk-jfx/tree/JDK-8227540-README) versus [kevinrushforth/jfx](https://github.com/kevinrushforth/jfx/tree/JDK-8227540-README) .

Notes to reviewers:

1. I pushed the proposed fix to my repo as three separate commits to make it easier to see the diffs. The first commit adds an exact copy of .github/README.md and .github/CONTRIBUTING.md to the top of the repo. The second commit modifies it as mentioned above. The third commit adds a link to javafx.io as suggested by PR #482. As always, they will be squashed into a single commit before anything is pushed to the repo.

2. The actual changeset that will be pushed to the HG repo will also remove the placeholder `README` file from the top of the repo. That placeholder file was manually removed from the `master` and `develop` branches of this sandbox some time ago, so it won't show up in this PR. Once this PR is approved, I will also post a webrev that will differ from this PR only in the removal of that dummy file via `hg rm README` .

### CODE REVIEW

**Comment by kevinrushforth** [comment]
> I think that the part at the very bottom about the OpenJDK rules ought to be mentioned
somewhere near the top

I'm not sure exactly which section you refer to. Do you mean the [Before submitting a pull request](https://github.com/javafxports/openjdk-jfx/blob/develop/.github/CONTRIBUTING.md#before-submitting-a-pull-request) sub-section in CONTRIBUTING.md? That is fairly near the beginning of the section about contributing changes. It possibly could be moved up slightly to the very beginning of the [Contributing code and documentation changes](https://github.com/javafxports/openjdk-jfx/blob/develop/.github/CONTRIBUTING.md#contributing-code-and-documentation-changes) section, but I wouldn't move it any further up than that (developers don't need an OCA to file a bug or enhancement).

Or did you mean this from the README.md file?

    OpenJFX is a project under the charter of the OpenJDK. The [OpenJDK Bylaws](http://openjdk.java.net/bylaws) and [License](LICENSE) govern our work. The OpenJFX project membership can be found on the [OpenJDK Census](http://openjdk.java.net/census#openjfx).

If so, then that will be seen prior to anything in `CONTRIBUTING.md`, since `README.md` is the one that is displayed on the front page. Developers never get to `CONTRIBUTING.md` without having seen `README.md` first.

Btw, the changes between the existing files in the `.github` directory and the copies added to the top directory are currently pretty small, and can be seen [here](https://github.com/javafxports/openjdk-jfx/pull/526/files/c44bd7245fdcf4fb4a37c66b1fb3879f00d08c71..1b9d3607fbba7d3ef138f5a92d039d6581c3d340).

**Comment by kevinrushforth** [comment]
I had a quick offline discussion with @prrace and he meant the section in the `README.md` file. Since that is displayed first, there should be no confusion.

However, one thing he did point out, which I will fix, is that the following is missing a link:

> See Community for details on how we work and how you can become a contributor.

I'll reword this, adding something similar to what Phil mentioned:

> "If this is your first time contributing to an OpenJDK project, you will want to review the rules [HERE], and sign up to be an OpenJDK contributor"

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+
+3. File a bug in JBS
+
+    If there isn't already a bug filed in [JBS](https://bugs.openjdk.java.net), then please file one at [bugreport.java.com](http://bugreport.java.com/). A JBS bug ID is needed even if you have already filed an issue in the GitHub issue tracker. A GitHub issue can be used as a convenience, but JBS is the official bug database for the OpenJFX Project. Once you submit a pull request, you can add the "github-bug" label on the JBS bug, along with an Issue link to the PR. If you don't have direct JBS access, one of the Project Committers or Authors will do this for you.
+
+4. Submit a pull request
+
+    Push your local changes to your forked copy of the repository and [submit a pull request](https://help.github.com/articles/using-pull-requests). In the pull request, choose a title which sums up the changes that you have made, and in the body provide more details about what your changes do. The title *should include a JBS Bug ID*, which will make it easier for reviewers to cross-reference your pull requests with JBS.
```
[the remark this answers is not in the document]
Comment (kevinrushforth) [reply CONTRIBUTING.md]: The redirected Wiki article looks right. Thanks for checking.

**Comment by brcolow** [comment]
Sorry to be somewhat off topic, but is project Skara a sure thing or are you only doing this work in case it is? Thanks.

**Comment by kevinrushforth** [comment]
> is project Skara a sure thing or are you only doing this work in case it is?

I can't guarantee that it is a sure thing, but we fully expect that it will happen. Now that Skara itself is open-sourced on GitHub, and using the Skara tooling for its own development (eating its own dog food so to speak), they are looking for some larger projects to go next, ahead of the jdk. OpenJFX would be a good candidate project given that many developers are already familiar with the GitHub workflow. If so, then once we get to the point where we are ready to present a transition plan and timeline, we would announce this on the mailing list and open it up for discussion.

**Pull request merged**